### PR TITLE
feat: add pre-commit with ansible-lint

### LINKED ISSUE #468: Enforce ansible-lint through a GitHub action
To avoid issues like missing spaces or indentation errors like in https://github.com/TOSIT-IO/tdp-collection/pull/459, we need to have a linter do the job for us at some point during the PR process.

### PR DESCRIPTION
**Pull request by leopaul36**
#### Which issue(s) this PR fixes

Fixes #468 

#### Additional comments

I created as draft because I am struggling to understand the behavior of `ansible-lint`.

I left on purpose an unnamed `block` task at `roles/hbase/phoenix/queryserver/daemon/tasks/kerberos.yml#L10` to explain.

Running `ansible-lint roles` (inside the venv) gives:

```
ARNING  The following filters were mocked during the run: Invalid plugin FQCN (tosit.tdp.access_fqdn): unable to locate collection tosit.tdp
You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental
  - name[missing]  # Rule for checking task and play names.

                              Rule Violation Summary                              
 count tag                       profile rule associated tags                     
    18 command-instead-of-shell  basic   command-shell, idiom (warning)           
     1 deprecated-command-syntax basic   command-shell, deprecations (warning)    
    26 key-order[task]           basic   formatting, experimental (warning)       
     1 name[missing]             basic   idiom                                    
     2 yaml[comments]            basic   formatting, yaml (warning)               
     3 yaml[indentation]         basic   formatting, yaml (warning)               
     7 yaml[line-length]         basic   formatting, yaml (warning)               
     3 risky-file-permissions    safety  unpredictability, experimental (warning) 
     5 risky-shell-pipe          safety  command-shell (warning)                  
     1 ignore-errors             shared  unpredictability, experimental (warning) 
    10 no-changed-when           shared  command-shell, idempotency (warning)     

Failed after min profile: 1 failure(s), 76 warning(s) on 400 files.
```

Whereas running just `ansible-lint` gives no failures:

```
WARNING  Replaced outdated tag 'unnamed-task' with 'name[missing]', replace it to avoid future regressions

Passed with production profile: 0 failure(s), 0 warning(s) on 680 files.
```

The issue is that the pre-commit hook for ansible-lint runs `python3 -m ansiblelint -vv --force-color` so in this case the task naming error is not catched for example.

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v3.2.0
+  hooks:
+   - id: trailing-whitespace
+   - id: end-of-file-fixer
+- repo: https://github.com/ansible/ansible-lint.git
+  rev: v6.8.7
```
Comment (gboutry) [inline .pre-commit-config.yaml:8]: ansible-lint 5.4.0 is the last version compatible python 3.6, I think we should stick with this one no ?

File: roles/hive/common/tasks/init_schema.yml [hunk 2]
```diff
@@ -2,33 +2,33 @@
...
-  run_once: true
-  become: true
-  become_user: "{{ hive_user }}"
-  changed_when: false
-  check_mode: false
-  register: hive_validate
-  failed_when: >-
-    hive_validate.rc > 1 or
-    (hive_validate.rc == 1 and 'Failed to get schema version' not in hive_validate.stderr)
+# - name: Validate Hive Metastore schema
```
Comment (gboutry) [inline roles/hive/common/tasks/init_schema.yml:5]: Was this forgotten ?

Comment (leopaul36) [reply roles/hive/common/tasks/init_schema.yml]: Yes my bad, force-pushed to fix

File: .pre-commit-config.yaml (hunk 1 above)
Comment (leopaul36) [reply .pre-commit-config.yaml]: Downgraded in the latest push force

**Comment by leopaul36** [comment]
So after downgrading to ansible-lint 5.4.0 with the latest push force (latest compatible with Ansible 2.9) I'm still having different behaviors...

`ansible-lint` returns with:

```
Finished with 34 failure(s), 4 warning(s) on 716 files.
```

`pre-commit` returns with:

```
Finished with 196 failure(s), 4 warning(s) on 716 files.
```

When running with pre-commit I can capture this process: `/home/leo/.cache/pre-commit/repo8_1inxp5/py_env-python3/bin/python /home/leo/.cache/pre-commit/repo8_1inxp5/py_env-python3/bin/ansible-lint --force-color`

At `/home/leo/.cache/pre-commit/repo8_1inxp5`, I am at commit `19efc81` of ansible-lint which corresponds to v5.4.0.

This makes me wonder if we really wan to enforce linting on the developper side, especially when the linting process is so long (git commit hangs for ~).

I'm starting to think that enforcing ansible-lint via a GitHub action would be enough. What do we think @gboutry @rpignolet @mehdibn @Nuttymoon ?

**Comment by gboutry** [comment]
It is true that the process is quite long, I would not mind enforcing it only via github action. Maybe a section on how to use it locally can be added

**Comment by leopaul36** [comment]
CLosing in favor of https://github.com/TOSIT-IO/tdp-collection/pull/597

**Pull request closed without merging**